### DISCUSSION TITLE
Idealy memory requirements when Harbor is deployed on K8S Cluster

### DISCUSSION THREAD

**Discussion by prashi2202** [goharbor/harbor#15763] [Q&A] [answered]
Hi Experts,

We have currently deployed Harbor on K8S cluster and would like to know from you all as to what is the ideal memory and CPU requirements, our target clients are expected to push large images which might go up to 100GB.

Would like to know what is the ideal memory and CPU configuration for such requirement.

Also need help on understanding how Jobservice, Core and Registry pods are interconnected with regards to memory management.  

Current configuration is as below.
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dspr/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dspr/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Pods | PROD - Memory   Limits in MB | CPU
-- | -- | --
nginx | 128 | 100m
portal | 128 | 100m
core | 512 | 100m
jobservice | 512 | 100m
registry | 3072 | 100m
registry-controller | 3072 | 100m
chartmuseum | 128 | 100m
notary | 128 | 100m
redis | 128 | 100m
exporter | 128 | 100m
trivy | 1024 | 200m
Total | 8960 | 1200m

</body>

</html>

**Comment by heww** [comment, marked as answer]
1200m means 1.2 core CPU, it's small for the normal usage of the harbor. 

The resource used by the harbor depends on the requests rate and network bandwidth, you can measure the resources using https://github.com/goharbor/perf

The registry uses a lot of memory when pulling/pushing large images.

**Reply by ErikEngerd** [reply]
Why does it need so much memory? I can imagine, pulling and pushing can be completely streamed. In fact, these nemory requirements are holding me back to use harbor.
